src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -158,19 +158,19 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
 
   // Components for collecting and generating output
 
-  import scala.reflect.internal.util.StatisticsInfo
+  import scala.reflect.internal.util.Statistics
   import scala.tools.nsc.transform.patmat.PatternMatchingStats
   trait GlobalStats extends ReflectStats
                        with TypersStats
                        with ImplicitsStats
                        with MacrosStats
                        with BackendStats
-                       with PatternMatchingStats
+                       with PatternMatchingStats { self: Statistics => }
```

Comment:
Because:
```
[error] [E1] src/compiler/scala/tools/nsc/Global.scala
[error]      illegal inheritance;
[error]       self-type Global.this.GlobalStats does not conform to Global.this.ReflectStats's selftype Global.this.ReflectStats with scala.reflect.internal.util.Statistics
[error]      L164:  trait GlobalStats extends ReflectStats
[error]      L164:                            ^
```
  
and 5 more errors like this one.
  
It wasn't obvious to me this was required. I would have expected the trait without the self-type annotation to compile...

Reply:
If the self-type is needed in `ReflectStats`, you could just make `GlobalStats extends Statistics`.

Reply:
Right, but that would require adding and forwarding the class parameters. I'm happy to keep it as it is right now.